### PR TITLE
(dsl): Support `matchBooleanPrefix` query

### PR DESCRIPTION
**Pull request by milicns**
Part of #91

### CODE REVIEW

File: modules/integration/src/test/scala/zio/elasticsearch/HttpExecutorSpec.scala [hunk 1]
```diff
@@ -917,6 +917,48 @@ object HttpExecutorSpec extends IntegrationSpec {
             Executor.execute(ElasticRequest.createIndex(firstSearchIndex)),
             Executor.execute(ElasticRequest.deleteIndex(firstSearchIndex)).orDie
           ),
+          test("search for a document using a match boolean prefix query") {
```
Comment (dbulaja98) [inline modules/integration/src/test/scala/zio/elasticsearch/HttpExecutorSpec.scala:920]: Write just one it-test for this query. There is no need to check `minimumShouldMatch` only.

**Review by dbulaja98 — CHANGES_REQUESTED** [review]

File: docs/overview/queries/elastic_query_match_boolean_prefix.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+val query: MatchBooleanPrefixQuery = matchBooleanPrefix(field = Document.stringField, value = "test")
+```
+
+If you want to change the `minimum_should_match` parameter, you can use the `minimumShouldMatch` method:
+```scala
+val queryWithMinimumShouldMatch: MatchBooleanPrefixQuery = matchBooleanPrefix(field = Document.stringField, value = "test").minimumShouldMatch(2)
+```
+
+You can find more information about `MatchBooleanPrefix` query [here](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/query-dsl-match-bool-prefix-query.html).
+
```
Comment (drmarjanovic) [inline docs/overview/queries/elastic_query_match_boolean_prefix.md:31]: Unnecessary new line.

File: modules/library/src/test/scala/zio/elasticsearch/ElasticQuerySpec.scala [hunk 3]
```diff
@@ -873,6 +873,57 @@ object ElasticQuerySpec extends ZIOSpecDefault {
             equalTo(Match[TestDocument, Double](field = "doubleField", value = 3.14))
           )
         },
+        test("matchBooleanPrefix") {
```
Comment (drmarjanovic) [inline modules/library/src/test/scala/zio/elasticsearch/ElasticQuerySpec.scala:876]: Move this between `matchAll` and `matches`.

File: modules/library/src/test/scala/zio/elasticsearch/ElasticQuerySpec.scala [hunk 4]
```diff
@@ -2427,6 +2478,76 @@ object ElasticQuerySpec extends ZIOSpecDefault {
           assert(queryTsInt.toJson(fieldPath = None))(equalTo(expectedTsInt.toJson)) &&
           assert(queryTsString.toJson(fieldPath = None))(equalTo(expectedTsString.toJson))
         },
+        test("matchBooleanPrefix") {
```
Comment (drmarjanovic) [inline modules/library/src/test/scala/zio/elasticsearch/ElasticQuerySpec.scala:2481]: Move this between `matchAll` and `matches`.

**Review by drmarjanovic — CHANGES_REQUESTED** [review]

**Review by dbulaja98 — APPROVED** [review]
Nice job!

**Pull request merged**